core/src/components/datetime/datetime.tsx

```diff
@@ -1196,14 +1196,22 @@ export class Datetime implements ComponentInterface {
     });
   }
 
-  private processValue = (value?: string | string[] | null) => {
-    const hasValue = value !== null && value !== undefined;
+  private processValue = async (value?: string | string[] | null, animate = false) => {
```

Comment:
Is the `animate` param needed here?

It sounds like we only want to animate when the value is change async. In which case, it should probably animate when the `reset` method is called too.

The only other time `processValue` is called is on `componentWillLoad` which is before the component is rendered. At that point, `elIsVisible` will be `false` anyways.

Reply:
Good point 🤔 I'll remove it.

Reply:
Discussed offline, but the problem here was the `elIsVisible` was `true` in `componentWillLoad` because it checked for `ion-datetime` visibility, not the calendar grid visibility. This caused the component to wait for scrolling to finish before the calendar body was visible, resulting in the scrolling to never happen.

Updating the check to `const elIsVisible = el.classList.contains('datetime-ready'); ` should fix the issue and allow us to remove the extra `animated` check.